internal/local/dependencies.go

```diff
@@ -61,5 +70,90 @@ func FindPackageJSON(path string) (Dependencies, error) {
 		return Dependencies{}, err
 	}
 
-	return Dependencies{rootDir, packageJSONPath}, nil
+	return Dependencies{rootDir, packageJSONPath, false}, nil
+}
+
+// PrepareUpload will prepare the dependencies for upload
+func (d Dependencies) PrepareUpload() (string, func(), error) {
```

Comment:
[q] Would it be beneficial to future devs to leave a short doc string/comment that the second argument is a cleanup function and should be called?

Reply:
yeah, i like that callout here